src/chrome/utils/typedOperators.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------
```

Comment:
Does this file get excluded from the lint rule? (just wondering how this works)

Reply:
It doesn't.
Instead of doing: !!stringOrUndefined we do: stringOrUndefined === undefined 
Instead of doing if (objectOrNull) we do: if (objectOrNull !== null)

All the code that appears on this file could appear directly on the code. I found it easier to replace
if (obj) for if (isNotEmpty(obj)) rather than if (obj !== undefined) because depending on the type if (obj) needs to be replaced by something different:
if obj is boolean? then you should do if (obj === true)
if obj is string?  then you should do if (obj !== undefined && obj !== '')
if obj is an object then you should do if (obj !== undefined)

Because the functions we are creating have specific types, if I try to do the wrong replacement, it won't compile. If I do it directly on the files, I won't get any warnings.

In some cases like hasMatches(match) I also find that more readable than match !== null

Reply:
Ah ok got it